### ISSUE TITLE
error  Parsing error: Unexpected character '@'

### ISSUE THREAD

**Issue by Songkunda** [eslint/eslint#10448]
hi,all.I had a problem using eslint@4.19.1;
environment:

 - eslint @4.19.1;
 - node @10.3.0;
 - npm @6.1.0
 - mobx @4.3.1
 - react @16.4.0

configuration:

  ```js
module.exports = {
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true,
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "ecmaVersion": 8,
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true,
      "jsx": true
    },
    "sourceType": "module"
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "indent": [
      "error",
      2
    ],
    "max-len": [
      "error", {
        "code": 120 //限制最大列数
      }
    ],
    "max-lines": ["error", 500],//限制最大行数
    "linebreak-style": [
      "error",
      "unix"
    ],
    // "quotes": [//强制使用一致的双引号
    //   "waring",
    //   "single"
    // ],
    "semi": [//不使用自动分号插入 (ASI)，强制使用一致的分号
      "error",
      "always"
    ],
    "semi-spacing": ["error", {"before": false, "after": true}],
    "no-unused-vars": [//检测变量定义后未使用
      "error",
      {
        "vars": "all",//检测变量
        "args": "all",//检测输入数组
        "caughtErrors": "all"//检测对于catch error 使用
      }
    ],
    "no-undef": [//检测变量未定义
      "error",
      {
        "typeof": true
      }
    ],
    "no-undef-init": "error",
    "no-console": [//阻止出现console调用
      "error",
      {
        "allow": ["warn", "error"]//排除warn 和error
      }
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always", //匿名函数需要前置空格
        "named": "never",//命名函数不需要前置空格
        "asyncArrow": "always"
      }],
    "unicode-bom": "error",//禁止BOM头
    "no-tabs": "error",//禁用tab
    "no-new-wrappers": "error",//禁用包装实例
    "no-control-regex": "error",//检测regex中出现控制符
    "no-new": "error",//禁止使用 new 关键字调用构造函数但却不将结果赋值给一个变量来保持一致性和约定
    "no-var": "error",//阻止 var 的使用，推荐使用 const 或 let
    "no-eval": ["error", {"allowIndirect": true}],//禁止直接调用eval
    "no-extend-native": "error",//禁止扩展原生类型
    "jsx-quotes": ["error"],//强制在 JSX 属性中使用一致的单引号或双引号 (默认是双引号)
    "require-await": "error",//对不包含 await 表达式的 async 函数发出警告
    "no-return-await": "error",//不在return中出现await
    "no-lone-blocks": "error",//消除脚本顶部或其它块中不必要的和潜在的令人困惑的代码块
    "sort-imports": ["error", {
      "ignoreCase": true,
      "ignoreMemberSort": false,
      "memberSyntaxSortOrder": ["none", "all", "multiple", "single"]
    }],//排序
    "sort-vars": ["error"],
    "valid-jsdoc": [
      "error",
      {
        "matchDescription": ".+",
        "requireReturn": false,
        "prefer": {//别名对应表
          "augments": "extends",
          "var": "member",
          "arg": "param",
          "argument": "param",
          "prop": "property",
          "func": "function",
          "method": "function",
          "desc": "description",
          "defaultvalue": "default"
        }
      }
    ]
  }
};

```

code : 

```js
import React, {Component} from "react";
import {observer} from "mobx-react";
@observer
class Header extends Component {
     constructor(props) {
    super(props);
    this.state = {
      "isBarVisibled": false
    }
  }
  render() {return <div/>}
}
```

```bash
 eslint ./src/components/header/index.jsx --debug
```

```bash
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in /Users/songkunda/flc/IFE-WebApps +0ms
  eslint:ignored-paths Could not find ignore file in cwd +0ms
  eslint:cli-engine Processing /Users/songkunda/flc/IFE-WebApps/src/components/header/index.jsx +0ms
  eslint:cli-engine Linting /Users/songkunda/flc/IFE-WebApps/src/components/header/index.jsx +1ms
  eslint:config Constructing config file hierarchy for /Users/songkunda/flc/IFE-WebApps/src/components/header +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /Users/songkunda/flc/IFE-WebApps/.eslintrc.js +1ms
  eslint:config-file Loading JS config file: /Users/songkunda/flc/IFE-WebApps/.eslintrc.js +0ms
  eslint:plugins Loaded plugin react (eslint-plugin-react@7.9.1) (from /usr/local/lib/node_modules/eslint-plugin-react/index.js) +0ms
  eslint:config-file Loading /usr/local/lib/node_modules/eslint/conf/eslint-recommended.js +183ms
  eslint:config-file Loading JS config file: /usr/local/lib/node_modules/eslint/conf/eslint-recommended.js +0ms
  eslint:config Using /Users/songkunda/flc/IFE-WebApps/.eslintrc.js +191ms
  eslint:config-ops Using config from partial cache +0ms
  eslint:config-ops Apply environment settings to config +1ms
  eslint:config-ops Creating config for environment browser +0ms
  eslint:config-ops Creating config for environment commonjs +1ms
  eslint:config-ops Creating config for environment es6 +0ms
  eslint:config-ops Creating config for environment node +0ms
  eslint:linter Linting code for /Users/songkunda/flc/IFE-WebApps/src/components/header/index.jsx (pass 1) +0ms
  eslint:linter Generating fixed text for /Users/songkunda/flc/IFE-WebApps/src/components/header/index.jsx (pass 1) +10ms
  eslint:text-fixer Applying fixes +0ms
  eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:cli-engine Linting complete in: 210ms +206ms

/Users/songkunda/flc/IFE-WebApps/src/components/header/index.jsx
  13:1  error  Parsing error: Unexpected character '@'

✖ 1 problem (1 error, 0 warnings)
```
How do I get him to know `@observer`？

**Comment by Songkunda** [comment]
My colleague told me how to do it ,like issues #4096
